extensions/binding/org.eclipse.smarthome.binding.bluetooth/src/main/java/org/eclipse/smarthome/binding/bluetooth/ConnectedBluetoothHandler.java

```diff
@@ -78,10 +78,14 @@ public void initialize() {
     public void dispose() {
         if (connectionJob != null) {
             connectionJob.cancel(true);
+            connectionJob = null;
         }
-        if (device != null) {
-            scheduler.submit(() -> device.disconnect());
-        }
+        scheduler.submit(() -> {
+            if (device != null) {
+                device.disconnect();
+                device = null;
```

Comment:
Do we care about threading?
`device` is set in the thread that calls `initialize` (in the parent class), `device` is unset (set to `null`) in a thread used by `scheduler`. So, should we use volatile, some synchronization, ...?

Reply:
You are right, I'll rework this...

Reply:
Using a reentrant lock now to make sure that different threads are correctly synchronized when accessing the device.